mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergFilterFactory.java

```diff
@@ -173,11 +172,11 @@ private static BigDecimal hiveDecimalToBigDecimal(HiveDecimalWritable hiveDecima
   }
 
   private static int daysFromDate(Date date) {
-    return DateTimeUtil.daysFromInstant(Instant.ofEpochMilli(date.getTime()));
+    return DateTimeUtil.daysFromDate(date.toLocalDate());
```

Comment:
I am having a hard time understanding what behavioral impact this change has. Can you elaborate a bit? Timestamp issues always tend to be tricky for me to get.

Reply:
Added extra comments to the methods, and in them I tried to describe how Hive parses literals.
Hope the comments are help to clarify the changes. If not, feel free to ask. It would be good to have good enough comments in the code, so any newcomer could understand the hows and the whys.

Thanks,
Peter

Reply:
Thanks for the comments! I understand now!